### ISSUE TITLE
Android app is crashing in development - JNI DETECTED ERROR IN APPLICATION

### ISSUE THREAD

**Issue by mydesweb** [facebook/react-native#34363]
### Description

Just upgraded react native to latest version and now the app is crashing because the app name contains diacritics
```
2022-08-08 00:42:43.156 28446-28513/ro.app A/tiefarabullyin: runtime.cc:669]   native: #14 pc 00000000002950d5  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (???)
    runtime.cc:669]   native: #15 pc 00000000002b1d99  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (facebook::flipper::FlipperConnectionManagerImpl::connectAndExchangeCertificate()+649)
    runtime.cc:669]   native: #16 pc 00000000002b12a4  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (facebook::flipper::FlipperConnectionManagerImpl::startSync()+596)
    runtime.cc:669]   native: #17 pc 00000000002bd421  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (???)
    runtime.cc:669]   native: #18 pc 00000000002b9923  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (void folly::detail::function::FunctionTraits<void ()>::callSmall<void folly::Executor::KeepAlive<folly::Executor>::add<folly::futures::detail::Core<folly::Unit>::doCallback(folly::Executor::KeepAlive<folly::Executor>&&, folly::futures::detail::State)::'lambda'(folly::Executor::KeepAlive<folly::Executor>&&)>(folly::futures::detail::Core<folly::Unit>::doCallback(folly::Executor::KeepAlive<folly::Executor>&&, folly::futures::detail::State)::'lambda'(folly::Executor::KeepAlive<folly::Executor>&&)&&) &&::'lambda'()>(folly::detail::function::Data&)+67)
    runtime.cc:669]   native: #19 pc 0000000000353e5a  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (folly::EventBase::FunctionLoopCallback::runLoopCallback()+10)
    runtime.cc:669]   native: #20 pc 000000000034fd9e  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (folly::EventBase::runLoopCallbacks()+222)
    runtime.cc:669]   native: #21 pc 0000000000350add  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (folly::EventBase::loopBody(int, bool)+813)
    runtime.cc:669]   native: #22 pc 00000000003519a6  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (folly::EventBase::loopForever()+22)
    runtime.cc:669]   native: #23 pc 000000000029f462  /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk!libflipper.so (offset 7fa000) (???)
    runtime.cc:669]   native: #24 pc 00000000003a032b  /apex/com.android.art/lib64/libart.so (art_quick_generic_jni_trampoline+219)
    runtime.cc:669]   native: #25 pc 000000000038c8c5  /apex/com.android.art/lib64/libart.so (nterp_helper+3837)
    runtime.cc:669]   native: #26 pc 00000000002ff958  [anon:dalvik-classes.dex extracted in memory from /data/app/~~WyoQbUFna2GHz35mK3wNag==/ro.app-FhwLXJn3VtDVxpFe3aSsTQ==/base.apk] (com.facebook.flipper.android.FlipperThread.run+40)
    runtime.cc:669]   native: #27 pc 0000000000395014  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+756)
    runtime.cc:669]   native: #28 pc 000000000041d9fa  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+218)
    runtime.cc:669]   native: #29 pc 000000000081aafe  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithJValues<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, jvalue const*)+478)
    runtime.cc:669]   native: #30 pc 000000000087a0cf  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1343)
    runtime.cc:669]   native: #31 pc 00000000000c758a  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+58)
2022-08-08 00:42:43.156 28446-28513/ro.app A/tiefarabullyin: runtime.cc:669]   native: #32 pc 000000000005fd87  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+55)
    runtime.cc:669]   at com.facebook.flipper.android.EventBase.loopForever(Native method)
    runtime.cc:669]   at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:31)
    runtime.cc:669] 
2022-08-08 00:42:43.156 28446-28513/ro.app A/tiefarabullyin: runtime.cc:677] JNI DETECTED ERROR IN APPLICATION: input is not valid Modified UTF-8: illegal continuation byte 0x25
    runtime.cc:677]     string: 'ws://localhost:9089?medium=1&sdk_version=4&app=App%20Name%20�%9A�%98&device_id=unknown&device=sdk_gphone64_x86_64%20-%2012%20-%20API%2031&os=Android'
    runtime.cc:677]     input: '0x77 0x73 0x3a 0x2f 0x2f 0x6c 0x6f 0x63 0x61 0x6c 0x68 0x6f 0x73 0x74 0x3a 0x39 0x30 0x38 0x39 0x3f 0x6d 0x65 0x64 0x69 0x75 0x6d 0x3d 0x31 0x26 0x73 0x64 0x6b 0x5f 0x76 0x65 0x72 0x73 0x69 0x6f 0x6e 0x3d 0x34 0x26 0x61 0x70 0x70 0x3d 0x41 0x70 0x70 0x25 0x32 0x30 0x4e 0x61 0x6d 0x65 0x25 0x32 0x30 0xc8 <0x25> 0x39 0x41 0xc8 0x25 0x39 0x38 0x26 0x64 0x65 0x76 0x69 0x63 0x65 0x5f 0x69 0x64 0x3d 0x75 0x6e 0x6b 0x6e 0x6f 0x77 0x6e 0x26 0x64 0x65 0x76 0x69 0x63 0x65 0x3d 0x73 0x64 0x6b 0x5f 0x67 0x70 0x68 0x6f 0x6e 0x65 0x36 0x34 0x5f 0x78 0x38 0x36 0x5f 0x36 0x34 0x25 0x32 0x30 0x2d 0x25 0x32 0x30 0x31 0x32 0x25 0x32 0x30 0x2d 0x25 0x32 0x30 0x41 0x50 0x49 0x25 0x32 0x30 0x33 0x31 0x26 0x6f 0x73 0x3d 0x41 0x6e 0x64 0x72 0x6f 0x69 0x64'
    runtime.cc:677]     in call to NewStringUTF
    runtime.cc:677]     from void com.facebook.flipper.android.EventBase.loopForever()
2022-08-08 00:42:43.157 28446-28513/ro.app A/libc: Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 28513 (FlipperEventBas), pid 28446 ()

```

### Version

0.69.4 

### Output of `npx react-native info`

System:
    OS: macOS 12.5
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Memory: 25.51 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.16.0/bin/yarn
    npm: 8.16.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
    Watchman: 2022.07.04.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.16 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.69.4 => 0.69.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Just change the app name `android/app/src/main/res/values/string.xml` to something like this
```
<resources>
    <string name="app_name">App Name ȚȘ</string>
</resources>
```

### Snack, code example, screenshot, or link to a repository

I think the problem is here
`JNI DETECTED ERROR IN APPLICATION: input is not valid Modified UTF-8:`

`'ws://localhost:9089?medium=1&sdk_version=4&app=App%20Name%20�%9A�%98&device_id=unknown&device=sdk_gphone64_x86_64%20-%2012%20-%20API%2031&os=Android'`

**Comment by cortinico** [comment]
Closing as this looks like a Flipper issue. 
It has been discussed/fixed here: https://github.com/facebook/flipper/issues/3175

We would need a flipper bump beyond 0.144.0 to fix for everyone.

You can solve the issue for the time being by just bumping your local version of Flipper on Android.

**Comment by cuitianze** [comment]
https://github.com/facebook/flipper/issues/3175#issuecomment-1112237452

android/gradle.properties
FLIPPER_VERSION=0.144.0

**Comment by mgscreativa** [comment]
I can confirm that this workaround fixes the issue (tested with flipper 0.144.0, 0.154.0 and 0.164.0):
